File: app/src/main/java/com/innerly/app/ui/MyGoalsActivity.java
package com.innerly.app.ui;

import android.content.Intent;
import android.database.Cursor;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.innerly.app.R;
import com.innerly.app.data.DatabaseHelper;
import com.innerly.app.data.Goal;
import com.innerly.app.utils.SessionManager;
import java.util.ArrayList;
import java.util.List;

public class MyGoalsActivity extends AppCompatActivity {

    private GoalAdapter adapter;
    private DatabaseHelper db;
    private SessionManager sessionManager;
    private RecyclerView rvGoals;
    private TextView tvEmptyState;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_my_goals);

        db = new DatabaseHelper(this);
        sessionManager = new SessionManager(this);

        rvGoals = findViewById(R.id.rvGoals);
        tvEmptyState = findViewById(R.id.tvEmptyState);
        Button btnNewGoal = findViewById(R.id.btnNewGoal);

        rvGoals.setLayoutManager(new LinearLayoutManager(this));

        adapter = new GoalAdapter(goal -> {
            Intent intent = new Intent(MyGoalsActivity.this, GoalDetailActivity.class);
            intent.putExtra("GOAL_ID", goal.getId());
            startActivity(intent);
        });
        rvGoals.setAdapter(adapter);

        btnNewGoal.setOnClickListener(v -> {
            startActivity(new Intent(MyGoalsActivity.this, CreateGoalActivity.class));
        });

        loadGoals();
    }

    @Override
    protected void onResume() {
        super.onResume();
        loadGoals();
    }

    private void loadGoals() {
        int userId = (int) sessionManager.getUserId();

        Cursor cursor = db.getGoalsForUser(userId);
        List<Goal> goals = new ArrayList<>();

        if (cursor != null) {
            if (cursor.moveToFirst()) {
                do {
                    Goal goal = new Goal();
                    int idIndex = cursor.getColumnIndex(DatabaseHelper.COL_GOAL_ID);
                    int titleIndex = cursor.getColumnIndex(DatabaseHelper.COL_GOAL_TITLE);
                    int contentIndex = cursor.getColumnIndex(DatabaseHelper.COL_GOAL_DESCRIPTION);

                    if (idIndex != -1) goal.setId(cursor.getLong(idIndex));
                    if (titleIndex != -1) goal.setTitle(cursor.getString(titleIndex));
                    if (contentIndex != -1) goal.setDescription(cursor.getString(contentIndex));

                    goals.add(goal);
                } while (cursor.moveToNext());
            }
            cursor.close();
        }

        if (goals.isEmpty()) {
            tvEmptyState.setVisibility(View.VISIBLE);
            rvGoals.setVisibility(View.GONE);
        } else {
            tvEmptyState.setVisibility(View.GONE);
            rvGoals.setVisibility(View.VISIBLE);
            adapter.setGoals(goals);
        }
    }
}

File: app/src/main/java/com/innerly/app/ui/MyGoalsActivity.kt
package com.innerly.app.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.innerly.app.R
import com.innerly.app.data.AppDatabase
import com.innerly.app.utils.SessionManager
import kotlinx.coroutines.launch

class MyGoalsActivity : AppCompatActivity() {
    private lateinit var adapter: GoalAdapter
    private lateinit var db: AppDatabase
    private lateinit var sessionManager: SessionManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_my_goals)

        db = AppDatabase.getDatabase(this)
        sessionManager = SessionManager(this)

        val rvGoals = findViewById<RecyclerView>(R.id.rvGoals)
        val tvEmptyState = findViewById<TextView>(R.id.tvEmptyState)
        val btnNewGoal = findViewById<Button>(R.id.btnNewGoal)

        rvGoals.layoutManager = LinearLayoutManager(this)
        adapter = GoalAdapter { goal ->
            val intent = Intent(this, GoalDetailActivity::class.java)
            intent.putExtra("GOAL_ID", goal.id)
            startActivity(intent)
        }
        rvGoals.adapter = adapter

        btnNewGoal.setOnClickListener {
            startActivity(Intent(this, CreateGoalActivity::class.java))
        }

        loadGoals()
    }

    override fun onResume() {
        super.onResume()
        loadGoals()
    }

    private fun loadGoals() {
        lifecycleScope.launch {
            val goals = db.goalDao().getActiveGoals(sessionManager.getUserId())
            if (goals.isEmpty()) {
                findViewById<TextView>(R.id.tvEmptyState).visibility = View.VISIBLE
                findViewById<RecyclerView>(R.id.rvGoals).visibility = View.GONE
            } else {
                findViewById<TextView>(R.id.tvEmptyState).visibility = View.GONE
                findViewById<RecyclerView>(R.id.rvGoals).visibility = View.VISIBLE
                adapter.setGoals(goals)
            }
        }
    }
}